lib/imsg-buffer.c

```diff
@@ -209,8 +209,10 @@ msgbuf_clear(struct msgbuf *msgbuf)
 {
 	struct ibuf	*buf;
 
-	while ((buf = TAILQ_FIRST(&msgbuf->bufs)) != NULL)
+	while ((buf = TAILQ_FIRST(&msgbuf->bufs)) != NULL) {
 		ibuf_dequeue(msgbuf, buf);
+		TAILQ_FIRST(&msgbuf->bufs) = NULL;
```

Comment:
Thanks @rwestphal, your explain is very clear, the `tqh_first` will be changed as the `TAILQ_REMOVE()` has been call and I have't not notice before.